### ISSUE TITLE
Mayavi in Jupyterlab

### ISSUE THREAD

**Issue by brey** [enthought/mayavi#733]
Hi all, does Mayavi has a compatible x3d extension for Jupyterlab?

**Comment by prabhuramachandran** [comment]
I am not sure, I haven't played with jupyterlab in a while or seen what it would take to support. Do you have any experience with this?

**Comment by mwaskom** [comment]
Piling onto this ... the ipy backend (which is delightful in the jupyter notebook) also does not seem to work in jupyterlab. There are no errors, but also nothing happens.

Jupyter lab now has a 1.0 release so there should be a stable path for adaptation. Unfortunately I do not know anything what would need to be done to make it happen :)

**Comment by mwaskom** [comment]
(Noting that supporting jupyterlab would be great for mayavi because the ability to create a separate view pane for the mayavi figure(s) works better with the more traditional workflow than the "linear" presentation in the original notebook).

**Comment by prabhuramachandran** [comment]
Yes, it should work with jupyterlab.  The latest ipyevents release fixed an issue allowing it to work. By this I mean the latest jupyter ipy backend, not the x3d backend.

**Comment by michaelaye** [comment]
test_plot3d() object is not interactive in JLab 2.0. new issue?

**Comment by brey** [comment]
> test_plot3d() object is not interactive in JLab 2.0. new issue?

Indeed
